include/fastrtps/rtps/common/Time_t.h

```diff
@@ -50,7 +50,22 @@ struct RTPS_DllAPI Time_t{
 		seconds = sec;
 		fraction = frac;
 	}
-	
+
+	Time_t(long double sec)
+	{
+		seconds = static_cast<uint32_t>(sec);
+		fraction = (sec - seconds) * 4294967296;
```

Comment:
I think this can produce a warning on some systems, as the result of the calculation is a long double. So I think a cast is required here.